### PR TITLE
Postprocessing Store

### PR DESCRIPTION
**Pull request by kobergj**
Add store functionality to postprocessing service

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiGraph-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/22317/51/1) failed. Further test are cancelled...

File: services/postprocessing/README.md [hunk 1]
```diff
@@ -14,6 +14,27 @@ The `postprocessing` service will then coordinate configured postprocessing step
...
+  -   `redis`: Stores data in a configured Redis cluster.
+  -   `redis-sentinel`: Stores data in a configured Redis Sentinel cluster.
+  -   `etcd`: Stores data in a configured etcd cluster.
+  -   `nats-js`: Stores data using key-value-store feature of [nats jetstream](https://docs.nats.io/nats-concepts/jetstream/key-value-store)
+  -   `noop`: Stores nothing. Useful for testing. Not recommended in production environments.
+
+1.  Note that in-memory stores are by nature not reboot-persistent.
+2.  Though usually not necessary, a database name and a database table can be configured for event stores if the event store supports this. Generally not applicable for stores of type `in-memory`. These settings are blank by default which means that the standard settings of the configured store apply.
+3.  Events stay in the store for 2 weeks by default. Use `POSTPROCESSING_RECORD_EXPIRY` to adjust this value.
+4.  The eventhistory service can be scaled if not using `in-memory` stores and the stores are configured identically over all instances.
```
[the remark this answers is not in the document]
Comment (kobergj) [reply services/postprocessing/README.md]: There is no `POSTPROCESSING_RECORD_EXPIRY`. Removed it.

File: services/postprocessing/README.md [hunk 2]
```diff
@@ -14,6 +14,24 @@ The `postprocessing` service will then coordinate configured postprocessing step
 
 When all postprocessing steps have completed successfully, the file will be made accessible for users.
 
+## Storing Postprocessing Data
+
+The `postprocessing` service needs to store some meta data about uploads to be able to orchestrate postprocessing. When running in single binary mode, the default in-memory implementation will be just fine. In distributed deployments it is recommended to use a persistent store, see below for more details.
```
Comment (EParzefall) [inline services/postprocessing/README.md:19]:
```suggestion
The `postprocessing` service needs to store some metadata about uploads to be able to orchestrate post-processing. When running in single binary mode, the default in-memory implementation will be just fine. In distributed deployments it is recommended to use a persistent store, see below for more details.
```

**Review by fschade — APPROVED** [review]
🚀

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**